
# Preprocessing

To get the salmon "expected counts", run the `run_salmon` script and spot check that the results 

# Load some libraries

```r
library(edgeR)
library(tximport)
```

# Start from a table of metadata

Read in the table of metadata and check that the files needed exist.

```r
md <- read.table("metadata.tsv", header=TRUE, stringsAsFactors=FALSE)
stopifnot(file.exists(md$salmonfile))
stopifnot(file.exists(md$bamfile))
```

# Use *tximport* to read in the quantifications

```r
txi <- tximport(structure(md$salmonfile, names = md$whatever), 
                type = “salmon”, txOut = TRUE)
```


# Setup edgeR container

We recognize, of course, that other reasonable options, such as sleuth/DESeq2/etc, exist.

```r
cts <- txi$counts
normMat <- txi$length
normMat <- normMat/exp(rowMeans(log(normMat)))
o <- log(calcNormFactors(cts/normMat)) + log(colSums(cts/normMat))

d <- DGEList(cts)
d$offset <- t(t(log(normMat)) + o)
d <- calcNormFactors(d)

mm <- model.matrix(~group,data=md)
```


# Always look at (multiple) diagnostics:  #1 multi-dimensional scaling plot

```r
plotMDS(d)
```

# Filter

We usually apply a filter to remove lowly expressed transcripts/genes.

```r
k <- rowSums(cpm(d)>1) >= 3
dim(d)
d <- d[k,]
dim(d)
```

# Dispersion estimation


```r
d <- estimateDisp(d,mm)
```

# Diagnostic: BCV plots

We generally observe a decreasing trend here.

```r
plotBCV(d)
```

# Setup for the result output

This is a little book-keeping, so that the output table has the transcript identifiers and the full table of counts-per-million.  Here is the place to add other useful stuff, such as gene symbols, chromosome locations, etc.

```r
cps <- cpm(d)
d$genes <- data.frame(transcript_id=rownames(d), 
                      round(cps,1))
```


# Formal statistical testing

Here, it is a simple two-group comparison, but more detailed comparisons can be easily made, using an appropriate combination of design matrix and contrasts/coefficients.

```r
f <- glmFit(d, mm)
lrt <- glmLRT(f, coef=2)
topTags(lrt)
```

# Diagnostic: Histogram of P-values

```r
hist(lrt$table$PValue, 50)
```


# Diagnostic: Spot check of some DE transcripts
```r
barplot( cps["ENST00000361886",] )
```


# Reproducibility

When you do these analyses, it is always a good idea to keep a record of the versions used

```r
sessionInfo()
```
